File: c/searching/linear_search.c
//
// Created by Vladimir-HP on 27/02/2021.
//
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define MAX 100
#define MIN -100


void fillArray(int array[], int n);

int linearSearch(int array[], int n, int x);


int main() {
    // Intializes random number generator
    srand(time(NULL));
    int n;
    printf("Number of elements in an array:");
    scanf("%d", &n);

    int array[n];
    fillArray(array, n);

    int x;
    printf("Enter the value that you are looking for:");
    scanf("%d", &x);

    if (linearSearch(array, n, x)) {
        printf("The value %d exists in generated array!", x);
    } else {
        printf("The value %d doesn't exist in generated array!", x);
    }

    exit(EXIT_SUCCESS);
}


void fillArray(int array[], int n) {
    for (int i = 0; i < n; ++i)
        array[i] = rand() % (MAX - MIN + 1) + MIN;
}


int linearSearch(int array[], int n, int x) {
    for (int i = 0; i < n; ++i)
        if (array[i] == x) return 1;

    return 0;
}

File: python/searching/linear_search.py
"""
 * @file   linear_search.py
 * @author Vladimir Mijic
 * @date   26/03/2021
"""


def linearSearch(array, x):
    n = len(array)

    for i in range(0, n):
        if array[i] == x:
            return True

    return False


if __name__ == '__main__':
    arr = [75, 41, 92, 60, 2, 0, 39, 23, 10, 68]
    print("Before:", arr)

    x = 23

    if linearSearch(arr, 23):
        print("The value %d exists in generated array!" % x)
    else:
        print("The value %d doesn't exist in generated array!", x)
